pkg/cloudprovider/providers/openstack/metadata.go

```diff
@@ -141,12 +151,27 @@ func getMetadataFromMetadataService() (*Metadata, error) {
 // Metadata is fixed for the current host, so cache the value process-wide
 var metadataCache *Metadata
 
-func getMetadata() (*Metadata, error) {
+func getMetadata(order string) (*Metadata, error) {
 	if metadataCache == nil {
-		md, err := getMetadataFromConfigDrive()
-		if err != nil {
-			md, err = getMetadataFromMetadataService()
+		var md *Metadata
+		var err error
+
+		elements := strings.Split(order, ",")
+		for _, id := range elements {
```

Comment:
Any difference with the previous code?  if `getMetadataFromConfigDrive` returns no error, `break` will be executed

Reply:
and that's fine, but now we can select the order in which `getMetadataFromConfigDrive` and `getMetadataFromMetadataService` are executed. In an env where we can't trust `getMetadataFromConfigDrive` or should be used as a secondary search option we can define so on `cloud.conf`.

Reply:
Ok. i see. So you mean adding another param to the openstack config file